mg/deps.go

```diff
@@ -104,10 +106,7 @@ func runDeps(ctx context.Context, fns ...interface{}) {
 
 func checkFns(fns []interface{}) {
 	for _, f := range fns {
-		switch f.(type) {
-		case func(), func() error, func(context.Context), func(context.Context) error:
-			// ok
-		default:
+		if !types.IsFuncType(f) {
```

Comment:
can you make this CheckFunc(f) and having it return an error? That way you wouldn't need to hard code the error text in every place we use this, it would just be part of the types package code.